City-of-Helsinki/drupal-helfi-platform-config
# UHF-10161: Create leijuke script for Telia ACE chat

# [UHF-10161](https://helsinkisolutionoffice.atlassian.net/browse/UHF-10161), [UHF-9328](https://helsinkisolutionoffice.atlassian.net/browse/UHF-9328) 

## What was done
* UHF-10161: Ports over [Telia's load_ace.js implementation](https://wds.ace.teliacompany.com/wds/instances/J5XKjqJt/load_ace.js) to a local Drupal library. Uses a modified version of the Chatleijuke scripts.
* UHF-9328: Adds cookie descriptions for Telia ACE implementations

## How to install
* Make sure your instance is up and running on latest dev branch. Recommended to use the [Palvelukeskus instance](https://github.com/City-of-Helsinki/drupal-palvelukeskus).
    * `git pull origin dev`
    * `make fresh`
* Update the Helfi Platform config
    * `composer require drupal/helfi_platform_config:dev-UHF-10161-telia-ace-script`
* Update HDBT
  * `composer require drupal/hdbt:dev-UHF-10161-telia-ace-script`  
* Run `drush locale:check && drush locale:update`
* Run `make drush-updb drush-cr`

## How to test
* [x] Check that the cookie descriptions appear and that they are correctly translated:
  * https://drupal-palvelukeskus.docker.so/fi/evasteasetukset
  * https://drupal-palvelukeskus.docker.so/en/cookie-information-and-settings
  * https://drupal-palvelukeskus.docker.so/sv/cookie-installningar
* [x] Add the Telia ACE Widget block to the Attachments region: https://drupal-palvelukeskus.docker.so/fi/admin/structure/block
  * Make sure the title is not visible
  * Chat Widget ID and translations: `matkapalvelut-chat_en`, `matkapalvelut-chat_fi` and `matkapalvelut-chat_sv`
  * Chat button label: "Matkapalvelut chat"
  * Add the block to `/helsingin-matkapalvelu`, `/helsingfors-resetjanst` and `/helsinki-transport-service` 
* Open this page in a private browsing window, only accept mandatory cookies: https://drupal-palvelukeskus.docker.so/fi/helsingin-matkapalvelu 
  * [x] Test the page in mobile mode. When you open the mobile menu, the placeholder button should be hidden and shouldn't show up on top of the menu. 
  * [x] You should see the chat button. No assets from wds.ace.teliacompany.com should be loaded yet and no chat cookies should be set yet
  * [ ] Click the button. It should be replaced with "Loading chat..." (translated to the correct language) until the chat widget loads.
  * [x] The chat widget should open automatically. On local environments the actual chat is most likely inoperable, but this should be fine. 
  * [x] Leave the widget open and reload the page. On the next page load it should open automatically again
  * [x] Close the widget and reload the page. On the next page load it should replace the placeholder button (after a brief load time) but the widget should remain closed. The icon might be correct (speech bubble icon) or might be bugged (caret icon), but we can't really affect this currently.
  * [x] Test the page in mobile mode. When you open the mobile menu, the chat button should be hidden and shouldn't show up on top of the menu.
* [x] Check the cookies, localstorage and sessionstorage data. Everything set by the Telia ACE implementation should be described in the cookie settings: https://drupal-palvelukeskus.docker.so/en/cookie-information-and-settings
  * Not all described cookies are currently in use, but the implementation should not set anything that isn't listed on the cookies page. 
* [x] Check that updated documentation makes sense: https://helsinkisolutionoffice.atlassian.net/wiki/spaces/HEL/pages/8426258434/Telia+ACE-widget

## Continuous documentation

## Translations

## Other PRs
* https://github.com/City-of-Helsinki/drupal-hdbt/pull/1014

[UHF-10161]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-10161?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[UHF-9328]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-9328?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
